apps/microtvm/zephyr/template_project/microtvm_api_server.py

```diff
@@ -162,7 +176,7 @@ def _get_device_args(options):
         return _get_openocd_device_args(options)
 
     raise BoardError(
-        f"Don't know how to find serial terminal for board {CMAKE_CACHE['BOARD']} with flash "
+        f"Don't know how to find serial terminal for board {_find_board_from_cmake_file(API_SERVER_DIR / CMAKELIST_FILENAME)} with flash "
```

Comment:
nit, and since it's out of scope for this PR I won't block on this (send a follow-on PR if you like): I think this error message should read something like:
`f"Don't know how to get device arguments for runner {flash_runner}"`  so I don't think mentioning the board here is really necessary, like it happens in 

https://github.com/apache/tvm/blob/main/apps/microtvm/zephyr/template_project/microtvm_api_server.py#L743

The runner is the important info here.